### ISSUE TITLE
Webhook when a new stream appears

### ISSUE THREAD

**Issue by sash2222** [nats-io/nats-server#2560]
## Feature Request
Webhook when a new stream appears
#### Use Case:
Our pain is in the dynamism, we need to know ahead of time whether some kind of flow has appeared.
The logic is this:
1. The producer created a stream and started posting messages there.
2. The consumer should find out about it, connect with the necessary parameters and read the messages
3. After emptying, the consumer must remove the flow and end.
Consumers are dedicated to each stream or subscription.
It would be ideal to do a webhook when a new topic appears, but it seems to me that it is not feasible.
#### Proposed Change:
As soon as JS "senses" that a new thread has appeared, it will immediately execute the WebHook with the flow parameters in the message. With this WebHuck we will understand that it is time to create a consumer.
#### Who Benefits From The Change(s)?
Anyone who cannot predict what to sign a consumer. Anyone with a theme or thread is very dynamic.
#### Alternative Approaches
The control queue is not very suitable, since the producer does not understand how often to send signals after what time to send them, etc.

**Comment by ripienaar** [comment]
If it’s for JetStream then I suggest you listen on the advisories instead of a web hook. 

Advisories tell you when streams are made/removed - and much more.

**Comment by sash2222** [comment]
Very similar to what I need.
I can catch the event $ JS.EVENT.ADVISORY.STREAM.CREATED. <STREAM> with my microservice and run it on the newly created consumer thread, then when I exhaust the stream, delete it and when I create the stream again, I will receive this event again.

**Comment by ripienaar** [comment]
Yea just be aware advisories - like webhooks - are not guaranteed delivery. 

So you should also periodically check with stream names API or similar.

**Comment by sash2222** [comment]
Yes, it will be something in the form of a workflow (temporal, cadence, etc.).
You can not close this thread yet, we will try, maybe there will be some questions.

**Comment by MauriceVanVeen** [comment]
Closing since there were no further questions. 

The `$JS.EVENT.ADVISORY.STREAM.CREATED.<STREAM>` advisory can be used to get notified of when a stream gets created.
https://docs.nats.io/jetstream/monitoring#advisories
